### PR TITLE
remove action plugin only fields from 'file' calls

### LINKED ISSUE #30556: Copy module with option "decrypt: no" causes error when file already exists
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
Module: Copy
Option: decrypt: no #new in version 2.4.

##### ANSIBLE VERSION

```
ansible 2.4.0.0
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

When copying a (on the remote host non-existing) ansible-vault encrypted file with the copy module everything works fine and the copied file is encrypted (as it should be).
When the file already exists however, an error is thrown.

##### STEPS TO REPRODUCE

Run the following task twice to reproduce

```yaml
---
- hosts: my-hosts
  tasks:
  - name: Copy vault-encrypted file
    copy:
      src: 'vault-encrypted.file'
      dest: .
      decrypt: no

  - name: Copy vault-encrypted file
    copy:
      src: 'vault-encrypted.file'
      dest: .
      decrypt: no
```

##### EXPECTED RESULTS

```
TASK [Copy vault-encrypted file] **********************************************************************************
changed: [my.server.com]

TASK [Copy vault-encrypted file] **********************************************************************************
ok: [my.server.com]
```

##### ACTUAL RESULTS

```
TASK [Copy vault-encrypted file] **********************************************************************************
changed: [my.server.com]

TASK [Copy vault-encrypted file] **********************************************************************************
fatal: [my.server.com]: FAILED! => {"changed": false, "checksum": "e88caee4fa8e6e905552d564e3d884097d8c67eb", "failed": true, "msg": "Unsupported parameters for (file) module: decrypt Supported parameters include: attributes,backup,content,delimiter,diff_peek,directory_mode,follow,force,group,mode,original_basename,owner,path,recurse,regexp,remote_src,selevel,serole,setype,seuser,src,state,unsafe_writes,validate"}
```

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixes #30556

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

copy
##### ANSIBLE VERSION

```
2.3/2.4/2.5
```

### CODE REVIEW

File: lib/ansible/plugins/action/copy.py [hunk 1]
```diff
@@ -186,6 +186,12 @@ def _recurse(topdir, rel_offset, parent_dirs, rel_base=u''):
 
 class ActionModule(ActionBase):
 
+    def _remove_action_keys(self, module_args):
```
Comment (abadger) [inline lib/ansible/plugins/action/copy.py]: Now that I can see this from a higher level, I think it makes sense to combine this filtering with creation of the args.

So this function becomes:
``` python
def _create_remote_copy_args(self, module_args):
    return dict((k, v) for k, v in module_args if k not in ('content', 'decrypt'))
```

And the calls to this get combined with the lines like:
``` python
- new_module_args = self._task.args.copy()
+ new_module_args = self._create_remote_copy_args(self._task.args)
```

Comment (bcoca) [reply lib/ansible/plugins/action/copy.py]: i was going even more generic, as this issue arises from 'remote file' args not matching

**Comment by abadger** [comment]
Should have an integration test case too (when file already exists but permissions are diff).  I'll add and push one.

**Comment by abadger** [comment]
Merged to devel and cherry-picked to stable-2.4 for the 2.4.1 release.

**Pull request merged**